src/status_im/contexts/chat/events.cljs

```diff
@@ -212,6 +212,11 @@
             #(when (group-chat? cofx chat-id)
                (loading/load-chat % chat-id))))
 
+(rf/reg-event-fx
+ :chat/clear-current-chat-id
```

Comment:
> This PR removes the key :current-chat-id when the chat screen is unmounted, by doing this, we are always able to navigate to a chat.

@ulisesmac I would need to dive deeper into the code to be more helpful.

It might be a good idea to start to question this concept of a global current chat state and gradually move away from it. If an action needs to be made related to a chat, if the ID was passed directly by callers, the ID would be ephemeral, like we do in most other parts of the code base and there would be no need to deal with state going stale. I don't know if my memory is serving me well, but I think it's not the first time, nor second that there are bugs related to the existence of a current chat ID.

Maybe this idea was necessary in the past and it's not the best now? Anyway, it's not an easy improvement, but could be done gradually. There are many references to the current chat ID in the repo.

Reply:
Yeah, there are lots of references to this value and IMO is hard to track and to keep in sync. The current bug is also related to keeping this value in sync, and we aren't able to have multiple chat screens in the navigation stack due to this value (e.g. when we press on a channel reference while being inside a chat, we don't navigate forward, we repaint the same screen with the data of the other chat)